### PR TITLE
[OASIS-12] Allow serializerexporter to consume APM stats

### PR DESCRIPTION
**Pull request by songy23**
### What does this PR do?

Add stats in channel to serializerexporter to consume APM stats
Split from https://github.com/DataDog/datadog-agent/pull/25759

### Motivation

Consume and send APM stats from DD connector.

### Describe how to test/QA your changes

n/a will be covered in https://github.com/DataDog/datadog-agent/pull/25759

### CODE REVIEW

File: comp/otelcol/otlp/components/exporter/datadogexporter/factory.go [hunk 1]
```diff
@@ -180,6 +190,38 @@ func (f *factory) createMetricsExporter(
...
+					}
+					for _, csp := range sp.Stats {
+						if csp.TracerVersion == "" {
+							csp.TracerVersion = tracerVersion
+						}
+					}
+					// The DD Connector doesn't set the agent version, so we'll set it here
+					sp.AgentVersion = agentVersion
+
+					// TODO(OASIS-12): send StatsPayload with trace agent
```
Comment (songy23) [inline comp/otelcol/otlp/components/exporter/datadogexporter/factory.go:218]: Will be added in https://github.com/DataDog/datadog-agent/pull/25759

File: comp/otelcol/otlp/components/exporter/datadogexporter/factory.go [hunk 2]
```diff
@@ -180,6 +190,38 @@ func (f *factory) createMetricsExporter(
 	return sf.CreateMetricsExporter(ctx, set, ex)
 }
 
+func (f *factory) consumeStatsPayload(ctx context.Context, statsIn <-chan []byte, tracerVersion string, agentVersion string, logger *zap.Logger) {
```
Comment (songy23) [inline comp/otelcol/otlp/components/exporter/datadogexporter/factory.go:193]: This is copied from https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/datadogexporter/factory.go#L254

File: comp/otelcol/otlp/components/exporter/datadogexporter/factory.go [hunk 3]
```diff
@@ -180,6 +189,38 @@ func (f *factory) createMetricsExporter(
 	return sf.CreateMetricsExporter(ctx, set, ex)
 }
 
+func (f *factory) consumeStatsPayload(ctx context.Context, wg *sync.WaitGroup, statsIn <-chan []byte, tracerVersion string, agentVersion string, logger *zap.Logger) {
```
Comment (songy23) [inline comp/otelcol/otlp/components/exporter/datadogexporter/factory.go:192]: This is copied from https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/datadogexporter/factory.go#L254

**Review by dineshg13 — APPROVED** [review]

**Pull request merged**